### DISCUSSION TITLE
Orientation by compass

### DISCUSSION THREAD

**Discussion by kds112** [devemux86/cruiser#235] [Q&A] [answered]
I am a basic Cruiser user. Cruiser is a big help for me when walking/hiking and you have to decide on going left or right at a fork. In older Cruiser versions when tapping on the compass you had the selections manual, compass, gps2d, gps3d available. Now the option "by compass" is gone. It would be very useful to offer this option again, the orientation in the countryside would be much easier for a hiker. Thanks a lot!

**Comment by devemux86** [comment]
@kds112 thanks for the suggestion.

It was disabled because phone compass usually is not very reliable and jumps without proper calibration.

If you find it useful, it can be enabled again in the next Cruiser update. :slightly_smiling_face:

**Reply by kds112** [reply]
That would be great! Thank you! Maybe a strong filter could be used for the compass information from the smartphone in order to avoid map jittering.

**Reply by 0709wiwiwi** [reply]
I once compared compass operation between Kurviger and Locus (**FILTER !**) with the phone mounted on the bike handlebar. 
The autoswitching from GPS to compass direction in Locus is triggered under a certain set speed. 
See the result in the next video: https://youtu.be/JwXrtD8mYD0
However, the compass had to be recalibrated too often to be reliable. I therefore did not use this any further. 
By the way, a track record, track used as a "tail", is also interesting as a direction reference when hiking.

**Reply by Leeb64** [reply]
Track logging is helpful in such a case

**Reply by kds112** [reply]
Normally I hike using just the trail signs. In uncertainty of the direction I switch on my phone, start cruiser, look for the right direction and switch off the phone again. Therefore the compass orientation feature would be very useful even if the compass direction has some deviation.
Second scenario, I stand on a hill looking around in the landscape and want to know the names of the villages/hills around. By using the compass orientation of the map it is really easy to handle. Thus I hope that the compass orientation feature comes back in Cruiser...

**Reply by devemux86** [reply]
Cruiser already includes a smooth mechanism for compass measurements.
We can improve it if you think it needs more processing.

**Reply by kds112** [reply]
I will check and give you a feedback when the feature is back again :-)

**Comment by devemux86** [comment, marked as answer]
Orientation by compass is again available in [Cruiser 3.0.25](https://play.google.com/store/apps/details?id=gr.talent.cruiser).

**Reply by kds112** [reply]
Thanks a lot! It works very well and the smooth mechanism is sufficient.

**Reply by devemux86** [reply]
@kds112 thanks for the feedback!
